### PR TITLE
Adding Files and Compiler.Common to projects to be packaged.

### PR DESCRIPTION
**Pull request by livarcocc**
Fixes https://github.com/dotnet/cli/issues/1755.

### CODE REVIEW

**Comment by livarcocc** [comment]
cc @brthor @victorhurdugaci @Sridhar-MS

**Comment by victorhurdugaci** [comment]
Thank you for the fast response! :)

File: scripts/package/projectsToPack.ps1 [hunk 1]
```diff
@@ -8,5 +8,7 @@ $ProjectsToPack = @(
     "Microsoft.DotNet.ProjectModel.Workspaces",
     "Microsoft.DotNet.InternalAbstractions",
     "Microsoft.Extensions.DependencyModel",
-    "Microsoft.Extensions.Testing.Abstractions"
+    "Microsoft.Extensions.Testing.Abstractions",
```
Comment (eerhardt) [inline scripts/package/projectsToPack.ps1]: (Really minor) Would it make sense to keep this list alphabetized?  That way we can quickly scan for any missed packages by looking at the folders under "src".

**Comment by eerhardt** [comment]
LGTM.

File: scripts/package/projectsToPack.ps1 (hunk 1 above)
Comment (livarcocc) [inline scripts/package/projectsToPack.ps1]: I re-ordered it.

**Comment by victorhurdugaci** [comment]
Btw, I didn't check but I only assumed those two were to only missing dependencies.

**Comment by brthor** [comment]
:+1:

**Comment by livarcocc** [comment]
@victorhurdugaci The only other one that is not here is the TestFramework. I am not sure what is up with it, so I left it out. If you want, I can add it as well.

File: scripts/package/projectsToPack.ps1 [hunk 2]
```diff
@@ -3,10 +3,13 @@
 
 $ProjectsToPack = @(
     "Microsoft.DotNet.Cli.Utils",
+    "Microsoft.DotNet.Compiler.Common",
+    "Microsoft.DotNet.Files",
```
Comment (davidfowl) [inline scripts/package/projectsToPack.ps1]: Wait seriously? Why?

Comment (livarcocc) [inline scripts/package/projectsToPack.ps1]: Because of https://github.com/dotnet/cli/issues/1755. ProjectModel.Workspaces depends on Compiler.Common.

**Comment by brthor** [comment]
:shipit:

File: scripts/package/projectsToPack.ps1 (hunk 2 above)
Comment (davidfowl) [inline scripts/package/projectsToPack.ps1]: We need to fix those dependencies. 

Microsoft.DotNet.Files shouldn't exist.
Microsoft.DotNet.Compiler.Common should only be used by the CLI.
Microsoft.DotNet.InternalAbstractions is an abomination that shouldn't have to leak to consumers. It's just for testing.

**Comment by livarcocc** [comment]
@dotnet-bot Test OSX Debug Build please
@dotnet-bot Test OSX Release Build please

**Comment by livarcocc** [comment]
@dotnet-bot Test OSX Debug Build please
@dotnet-bot Test OSX Release Build please

**Comment by livarcocc** [comment]
@dotnet-bot Test this please.

**Comment by livarcocc** [comment]
@dotnet-bot Test this please.

**Pull request closed without merging**